src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -239,6 +239,8 @@ namespace winrt::TerminalApp::implementation
         });
         _root->Create();
 
+        AppLogic::Current()->SettingsChanged({ get_weak(), &TerminalWindow::UpdateSettingsHandler });

```

Comment:
If I'm not mistaken, this trades one theoretical problem for another one though... Now it's possible to miss a settings update in the time between `TerminalWindow` creation and startup. But that's better than crashing so I'm convinced we should just merge this. 😅

I think one way to fix this would be to not pass any settings to the `TerminalWindow` constructor and instead have `TerminalWindow` pull the settings from `AppLogic` the same way you ask it for `SettingsChanged` events now. If we do that after calling `SettingsChanged()`, it'll be impossible to miss any updates, because `UpdateSettings` yields to the UI thread and thus must run after `Create()` returns.

We can also implement other options of course, since reaching back into `AppLogic` feels a little sneaky, similar to `ServiceLocator`. Like delaying the update until we flag a `std::latch` to signal that updating the settings is now safe, etc. If you want to solve this at some point and need help, let me know!

Until then I personally think it would be nice to note this down as a code comment there, so we don't forget about it. We can make it a github issue, but idk... it's like a P5 bug now.